src/settings.rs

```diff
@@ -94,38 +97,78 @@ pub fn load_from_flash<T: for<'d> JsonCoreSlash<'d, Y>, const Y: usize>(
             _ => continue,
         };
 
+        // An empty vector may be saved to flash to "erase" a setting, since the H7 doesn't support
+        // multi-write NOR flash. If we see an empty vector, ignore this entry.
+        if item.0.is_empty() {
+            continue;
+        }
+
         log::info!("Loading initial `{path}` from flash");
 
-        let flavor = postcard::de_flavors::Slice::new(&item.data);
-        if let Err(e) =
-            structure.set_postcard_by_key(path.split('/').skip(1), flavor)
+        let mut deserializer = postcard::Deserializer::from_flavor(
+            postcard::de_flavors::Slice::new(&item.0),
+        );
+        if let Err(e) = structure
+            .deserialize_by_key(path.split('/').skip(1), &mut deserializer)
```

Comment:
Why can't we `set_postcard_by_key` etc?

Reply:
Yeah those got deleted by an incorrect `git merge` process. Fixed it in 48a824c4d380550d920fb2c5caf19a5025512e03